core/src/main/java/com/linecorp/armeria/server/ServerSslContextUtil.java

```diff
@@ -94,13 +98,26 @@ static SSLSession validateSslContext(SslContext sslContext) {
         return sslSession;
     }
 
+    private static TlsEngineType determineTlsEngineType(SslContext sslContext) {
```

Comment:
Determining which `TlsEngineType` should be matched is based on  `io.netty.handler.ssl.SslContext#newServerContextInternal`.

```java 
        switch (provider) {
        case JDK:
            if (enableOcsp) {
                throw new IllegalArgumentException("OCSP is not supported with this SslProvider: " + provider);
            }
            return new JdkSslServerContext(sslContextProvider,
                    trustCertCollection, trustManagerFactory, keyCertChain, key, keyPassword,
                    keyManagerFactory, ciphers, cipherFilter, apn, sessionCacheSize, sessionTimeout,
                    clientAuth, protocols, startTls, keyStoreType);
        case OPENSSL:
            verifyNullSslContextProvider(provider, sslContextProvider);
            return new OpenSslServerContext(
                    trustCertCollection, trustManagerFactory, keyCertChain, key, keyPassword,
                    keyManagerFactory, ciphers, cipherFilter, apn, sessionCacheSize, sessionTimeout,
                    clientAuth, protocols, startTls, enableOcsp, keyStoreType, ctxOptions);
        case OPENSSL_REFCNT:
            verifyNullSslContextProvider(provider, sslContextProvider);
            return new ReferenceCountedOpenSslServerContext(
                    trustCertCollection, trustManagerFactory, keyCertChain, key, keyPassword,
                    keyManagerFactory, ciphers, cipherFilter, apn, sessionCacheSize, sessionTimeout,
                    clientAuth, protocols, startTls, enableOcsp, keyStoreType, ctxOptions);
        default:
            throw new Error(provider.toString());
        }
```

Reply:
Should we use `TlsEngineType` specified in `VirtualHost`?
It would be better to just use the given option rather than try to guess.

Reply:
I've used `tlsEnginType` in `VirtualHost` to determine it. Thanks for the suggestion!